cpp/export_vufind_redirects.cc

```diff
@@ -0,0 +1,48 @@
+/** \brief Tool to export all datasets from VuFind redirect table to CSV file.
+ *  \author Mario Trojan
+ *
+ *  \copyright 2020 Universitätsbibliothek Tübingen.  All rights reserved.
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Affero General Public License as
+ *  published by the Free Software Foundation, either version 3 of the
+ *  License, or (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+#include <iostream>
+#include <cstdlib>
+#include "File.h"
+#include "FileUtil.h"
+#include "TextUtil.h"
+#include "VuFind.h"
+#include "util.h"
+
+
+int Main(int argc, char **argv) {
+    if (argc != 2)
+        ::Usage("export_file");
+
+    const auto db_connection(VuFind::GetDbConnection());
+    db_connection->queryOrDie("SELECT * FROM tuefind_redirect");
+    auto result_set(db_connection->getLastResultSet());
+
+    File csv_file(argv[1], "w", File::ThrowOnOpenBehaviour::THROW_ON_ERROR);
+    csv_file.write("url;group;timestamp\n");
+    while (const auto &db_row = result_set.getNextRow()) {
+        std::string csv_row;
+        csv_row += TextUtil::CSVEscape(db_row["url"]) + ";";
+        csv_row += TextUtil::CSVEscape(db_row["group_name"]) + ";";
+        csv_row += TextUtil::CSVEscape(db_row["timestamp"]) + "\n";
+        csv_file.write(csv_row);
+    }
+    csv_file.close();
```

Comment:
You don't need this line.  The destructor of the File class already takes care of it.

Reply:
removed